sdk/python/feast/entity.py

```diff
@@ -44,3 +56,246 @@ def from_proto(cls, entity_proto: EntityProto):
         """
         entity = cls(name=entity_proto.name, dtype=ValueType(entity_proto.value_type))
         return entity
+
+
+class EntityV2:
+    """
+    Represents a collection of entities and associated metadata.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        value_type: ValueType,
+        labels: Optional[MutableMapping[str, str]] = None,
+    ):
+        self._name = name
+        self._description = description
+        self._value_type = value_type
+        if labels is None:
+            self._labels = OrderedDict()  # type: MutableMapping[str, str]
+        else:
+            self._labels = labels
+
+        self._created_timestamp: Optional[Timestamp] = None
+
+    def __eq__(self, other):
+        if not isinstance(other, EntityV2):
+            return NotImplemented
+
+        if isinstance(self.value_type, int):
+            self.value_type = ValueType(self.value_type).name
+        if isinstance(other.value_type, int):
+            other.value_type = ValueType(other.value_type).name
+
+        if (
+            self.labels != other.labels
+            or self.name != other.name
+            or self.description != other.description
+            or self.value_type != other.value_type
+        ):
+            return False
+
+        return True
+
+    def __str__(self):
+        return str(MessageToJson(self.to_proto()))
+
+    @property
+    def name(self):
+        """
+        Returns the name of this entity
+        """
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        """
+        Sets the name of this entity
+        """
+        self._name = name
+
+    @property
+    def description(self):
+        """
+        Returns the description of this entity
+        """
+        return self._description
+
+    @description.setter
+    def description(self, description):
+        """
+        Sets the description of this entity
+        """
+        self._description = description
+
+    @property
+    def value_type(self):
+        """
+        Returns the type of this entity
+        """
+        return self._value_type
+
+    @value_type.setter
+    def value_type(self, value_type: ValueType):
+        """
+        Set the type for this entity
+        """
+        self._value_type = value_type
+
+    @property
+    def labels(self):
+        """
+        Returns the labels of this entity. This is the user defined metadata
+        defined as a dictionary.
+        """
+        return self._labels
+
+    @labels.setter
+    def labels(self, labels: MutableMapping[str, str]):
+        """
+        Set the labels for this entity
+        """
+        self._labels = labels
+
+    @property
+    def created_timestamp(self):
+        """
+        Returns the created_timestamp of this entity
+        """
+        return self._created_timestamp
+
+    @created_timestamp.setter
+    def created_timestamp(self, created_timestamp):
```

Comment:
Why do we expose a setter for this meta field?

Reply:
Removed.